root/user/UserTagEntity.js

```diff
@@ -0,0 +1,68 @@
+/*
+ * @flow
+ * Copyright (C) 2019 MetaBrainz Foundation
+ *
+ * This file is part of MusicBrainz, the open internet music database,
+ * and is licensed under the GPL version 2, or (at your option) any
+ * later version: http://www.gnu.org/licenses/gpl-2.0.txt
+ */
+
+import * as React from 'react';
+
+import UserAccountLayout from '../components/UserAccountLayout';
+import {EntityListContent} from '../tag/EntityList';
+
+import DownvotedSwitch from './components/DownvotedSwitch';
+import UserHasNotUsedTag from './components/UserHasNotUsedTag';
+import UserTagHeading from './components/UserTagHeading';
+
+type UserTagEntityProps = {
+  +$c: CatalystContextT,
+  +entityTags: $ReadOnlyArray<{
+    +entity: CoreEntityT,
+    +entity_id: number,
+  }>,
+  +entityType: string,
+  +page: string,
+  +pager: PagerT,
+  +showDownvoted: boolean,
+  +tag: TagT,
+  +user: EditorT,
+};
+
+const UserTagEntity = ({
+  entityTags,
+  entityType,
+  page,
+  pager,
+  showDownvoted,
+  tag,
+  user,
+}: UserTagEntityProps) => (
+  <UserAccountLayout entity={user} page={page}>
+    <UserTagHeading showDownvoted={showDownvoted} tag={tag} />
+
+    <DownvotedSwitch
+      showDownvoted={showDownvoted}
+      tag={tag}
+      user={user}
+    />
+
+    {entityTags.length > 0 ? (
+      <EntityListContent
+        entityTags={entityTags}
+        entityType={entityType}
+        pager={pager}
+      />
+    ) : (
+      <UserHasNotUsedTag
+        entityType={entityType}
+        showDownvoted={showDownvoted}
+        tag={tag}
+        user={user}
+      />
```

Comment:
Is saying "user has not tagged anything with foo" or "user has not voted against any foo tags" here when they haven't with this particular entity type but *have* with others a bit misleading? At least it appears under e.g. "0 artists found" so it has some context, but dunno if that applies as much for other translations. It might be clearer with only the heading, actually ("0 artists found") and no explanatory text.

Reply:
I think the explanatory text makes sense, but I made it so that it changes based on entity.